File: src/astek.rs
use chrono::{DateTime, Local};
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
struct Indisponibility {
    start: DateTime<Local>,
    end: DateTime<Local>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct Astek {
    name: String,
    indisponibilities: Vec<Indisponibility>,
}

File: frontend/src/types/astek.ts
import { UUID } from "crypto";

export type Astek = {
  id: UUID;
  indisponibilities: Indisponibility[];
};

export enum IndisponibilityType {
  Private = "Private",
}

export type Indisponibility = {
  type: IndisponibilityType;
  start: Date;
  end: Date;
};
